src/os/ui/menu/defaultwindowsmenu.js

```diff
@@ -33,20 +45,23 @@ os.ui.menu.windows.default.setup = function() {
     'html': 'adddata'
   }, true);
 
+  const layersDefaults = /** @type {osx.window.WindowOptions} */ (
+    settings.get(os.ui.menu.windows.default.SettingsKey.LAYERS_DEFAULTS, {}));
+
   var layers = os.ui.menu.windows.addWindow('layers', {
     'key': 'layers',
-    'icon': 'fa fa-layer-group',
-    'label': 'Layers',
+    'icon': layersDefaults.icon || 'fas fa-layer-group',
+    'label': layersDefaults.label || 'Layers',
     'description': 'View and manipulate layers on the map',
-    'x': '0',
-    'y': '96',
-    'width': '350',
-    'height': '550',
-    'min-width': '300',
-    'max-width': '1000',
-    'min-height': '250',
-    'max-height': '2000',
-    'show-close': 'true',
+    'x': layersDefaults.x != null ? layersDefaults.x : 12,
+    'y': layersDefaults.y != null ? layersDefaults.y : 96,
+    'width': layersDefaults.width != null ? layersDefaults.width : 400,
+    'height': layersDefaults.height != null ? layersDefaults.height : 550,
+    'min-width': layersDefaults.minWidth != null ? layersDefaults.minWidth : 300,
+    'max-width': layersDefaults.maxWidth != null ? layersDefaults.maxWidth : 1000,
+    'min-height': layersDefaults.minHeight != null ? layersDefaults.minHeight : 350,
+    'max-height': layersDefaults.maxHeight != null ? layersDefaults.maxHeight : 2000,
+    'show-close': layersDefaults.showClose != null ? layersDefaults.showClose : true,
```

Comment:
Might be a bit cleaner to have a default object with all the hardcoded values in it and just merge the object from settings into it.

Reply:
Yeah that's a better idea. Initially I had a few values here, and fell into the slippery slope of adding more things. I'll merge instead.

Reply:
Fixed that and rebased to resolve a merge conflict.